reporter/doc.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 Mark Nevill. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+/*
+`reporter` provides client and server side reporting of HTTP request and response for stats.
```

Comment:
this entire doc still reads like it's for metrics. the reporter doesn't do anything with stats, its provides callback for particular request-processing events

Reply:
Well it is `for stats` because it enables stats (: Will rephrase.